src/index.ts

```diff
@@ -81,10 +106,56 @@ class Bot {
   public async deinit(): Promise<void> {
     // Stop the clients first, so that they aren't trying to
     // talk to a deinit'ed service
+    if (this._initStatus === 'deinitializing' || this._initStatus === 'deinitialized') {
+      this.debugLog('Trying to deinitialize, but already called', 'I')
+    } else {
+      this._initStatus = 'deinitializing'
+      this.debugLog('beginning deinit')
+      await this.chat._deinit()
+      await this._service.deinit()
+      await rmdirRecursive(this._workingDir)
+      this.debugLog('finished deinit')
+      this._initStatus = 'deinitialized'
+    }
+  }
+
+  /**
+   * If bot is initialized with an optional directory `adminDebugDirectory`, this will write that string into a file in there
+   * This function is also called automatically when in adminDebug mode, and will record plaintexts of messages into that directory,
+   * so use wisely.
+   * @memberof Bot
+   * @example
+   * bot.log("My bot is ready to go.")
+   */
+  public async debugLog(text: string, infoOrErr?: 'I' | 'E'): Promise<void> {
+    if (this.adminDebugDirectory) {
+      const code = infoOrErr ? infoOrErr : 'I'
+      const line = `${new Date().toISOString()} [${code}] ${text}\n`
```

Comment:
We should use `os.EOL` here for windows compatibility. https://nodejs.org/api/os.html#os_os_eol

Reply:
on it